x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/RestExplainLifecycleAction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.indexlifecycle.action;
+
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.core.indexlifecycle.action.ExplainLifecycleAction;
+import org.elasticsearch.xpack.indexlifecycle.IndexLifecycle;
+
+import java.io.IOException;
+
+public class RestExplainLifecycleAction extends BaseRestHandler {
+
+    public RestExplainLifecycleAction(Settings settings, RestController controller) {
+        super(settings);
+        controller.registerHandler(RestRequest.Method.GET, "_" + IndexLifecycle.NAME + "/explain", this);
+        controller.registerHandler(RestRequest.Method.GET, "{index}/_" + IndexLifecycle.NAME + "/explain", this);
```

Comment:
> interesting. we are bypassing the original base-path that is namespaced by xpack?

Yes, this is consistent with rollup which exposes an `{index}/_rollup_search` endpoint

> I would like the location of {index} in our APIs to be consistent. Not sure what exists now in RetryAction is
the best, but consistency is more important for me.

Agreed that it should be consistent. I like having `{index}` first because the action name (e.g. in `ExplainLifecycleAction`) is an `indices:*` action so it make sense that all `indices:*` action's URLs start with the index.

Reply:
I didn't know x-pack is OK with supporting things before the _xpack namespace... sounds good to me! I will revisit my endpoint